services/cd-service/pkg/repository/repository.go

```diff
@@ -2512,16 +2512,15 @@ func (s *State) UpdateTopLevelAppInOverview(ctx context.Context, transaction *sq
 		}
 		rels = retrievedReleasesOfApp
 	}
-	for _, id := range rels {
-		if rel, err := s.GetApplicationRelease(ctx, transaction, appName, id); err != nil {
-			return err
-		} else {
+
+	if releasesInDb, err := s.GetApplicationReleasesDB(ctx, transaction, appName, rels); err != nil {
```

Comment:
Not sure, but shouldn't it also work for the case when we are not using the db?

Reply:
this functions is only called from the db package, so this code should never run without db.

Reply:
nevermind, i added it, the test still call this function without the DB enabled